queries/planet_osm_polygon.jinja2

```diff
@@ -133,7 +133,7 @@ FROM (
        mz_poi_min_zoom IS NOT NULL OR
        mz_transit_level IS NOT NULL OR
        mz_water_min_zoom IS NOT NULL))
-{% else %}
+{% elif zoom >= 8 %}
```

Comment:
Can you confirm that this limit OSM **boundary** layer related features to zoom >= 8? Are there any other changes needed to support that?

**boundaries** wasn't in the original list of affected layers, but looks like we set the OSM country features to min_zoom 8 in the YAML, but the "same" NE features get min_zoom 0 right now. They should be consistent, and while we shouldn't change that min_zoom in this PR we should allow for them to be equal but "transition" to the OSM feature at zoom 8.